core/variant/type_info.h

```diff
@@ -367,4 +367,75 @@ ZERO_INITIALIZER_NUMBER(char32_t)
 ZERO_INITIALIZER_NUMBER(float)
 ZERO_INITIALIZER_NUMBER(double)
 
+template <typename T>
+inline constexpr bool is_variant_type_v = TypeInherits<Object, T>::value;
+
+template <typename T, typename = void>
+struct GetVariantType;
+
+template <typename T>
+struct GetVariantType<T, typename EnableIf<TypeInherits<Object, T>::value>::type> {
+	static const Variant::Type value = Variant::OBJECT;
+};
+
+template <typename T>
+inline constexpr Variant::Type get_variant_type_v = GetVariantType<T>::value;
+
+#define MAKE_VARIANT_TYPE_CONSTANTS(m_type, m_variant_type) \
```

Comment:
Isn't there a way to get the `Variant::Type` from a type like `bool` or `String`? It would reduce the number of arguments needed in this macro.

Reply:
There's a function for getting the type from the Variant class, but I don't *think* one exists for the type itself; not in a `constexpr` context anyway

Reply:
If there isn't, maybe it would be appropriate to make one in this PR? I think you could do it with templates like
```cpp
template<typename T>
Variant::Type GetBuiltinType() { return Variant::MAX; }

template<Variant>
Variant::Type GetBuiltinType() { return Variant::NIL; }

template<bool>
Variant::Type GetBuiltinType() { return Variant::BOOL; }

// etc...
```
If something like this doesn't already exist somewhere, I'm sure it would be a useful utility to have around.

Reply:
Unless I'm misunderstanding what you're suggesting, that *is* what this PR is implementing; `get_variant_type_v` is exactly that.

Reply:
oh, duh, of course xD. I glossed over what the `MAKE_VARIANT_TYPE_CONSTANTS` macro was actually doing and now I realize it is literally doing exactly what I suggested 🤣 .

I guess the only different is whether it's a template function `GetBuiltinType<T>` or a template `constexpr`. I feel like I'm not qualified enough to know the difference, but my sense is that the template function approach is slightly more consistent with the style in some of the other variant files. That said, that might just be because `constexpr` didn't exist when that other code was written 🤷

Reply:
I'm honestly just following the file's above implementations using `constexpr`. There's probably a broader discussion to be had on the formatting of `constexpr` in standalone values, but that's well outside the scope of this PR.